apps/blaze-dashboard/src/routes.js

```diff
@@ -31,13 +32,23 @@ export default function ( pageBase = '/' ) {
 
 	blazePage( getAdvertisingDashboardPath( '/:site' ), promotedPosts );
 
-	blazePage( getAdvertisingDashboardPath( '/:site/promote/:item' ), promoteWidget );
+	blazePage(
+		getAdvertisingDashboardPath( '/:tab/:site' ),
+		checkValidTabInNavigation,
+		promotedPosts
+	);
 
-	blazePage( getAdvertisingDashboardPath( '/:site/:tab' ), promotedPosts );
+	blazePage( getAdvertisingDashboardPath( '/campaigns/:campaignId/:site' ), campaignDetails );
 
-	blazePage( getAdvertisingDashboardPath( '/:site/:tab/promote/:item' ), promoteWidget );
+	blazePage( getAdvertisingDashboardPath( '/promote/:item/:site' ), promoteWidget );
 
-	blazePage( getAdvertisingDashboardPath( '/:site?/campaigns/:campaignId' ), campaignDetails );
+	blazePage( getAdvertisingDashboardPath( '/:tab/promote/:item/:site' ), promoteWidget );
+
+	// Compatibility: Redirects to new navigation style
+	page( getAdvertisingDashboardPath( '/:site/:tab/promote/:item' ), ( { params } ) => {
+		const { site, tab, item } = params;
+		page.redirect( getAdvertisingDashboardPath( `/${ tab }/promote/${ item }/${ site }` ) );
+	} );
```

Comment:
I have a thought about marking it as used for compatibility.

I noticed that there two ways of opening a "Ready to Promote" posts tab:
- `/advertising/posts/:site`, and
- `/advertising/:site` (default tab).

If we plan to remove compatibility path support, we should change to URL for posts tab (when user clicks to tab manually) to `/advertising/:site`.

Just thinking out loud. At the moment everything works like a charm.

Reply:
Thanks for considering this, @sbarbosa!

I'm totally fine to leave everything as-is. Feel free to merge.